### PR TITLE
[BCI-1263] Generalised Headtracker config

### PR DESCRIPTION
**Pull request by ExtremelySunnyYK**
This PR aims to create a wrapper for HeadTracker Config to generalise Headtracker Config. 

**Motivation:**
- HeadTracker config is important when creating a headtracker. However the current Headtracker config is EVM specific in its naming.
- By creating a Headtracker Config wrapper, the existing evm config could still make use of the underlying structure of the evm Config, but with changes to naming to make it chain agnostic. 
- We are using wrapper pattern to reduce propagation of changes deep into the codebase.

**Changes:**
- The chain-agnostic Config is created in the common folder.
- Wrapper function for EVM HeadTrackerConfig is created

**Note:**
- This PR follows a similar pattern of using wrapper as [previous PR of generalising TXM config](https://github.com/smartcontractkit/chainlink/pull/8932/files#diff-842bf192dd563c110d856a44655696823d2bf8fd923c6b7f4afe10bbb5050fd5).

**References:**
- [Ticket](https://smartcontract-it.atlassian.net/browse/BCI-1263)
- [HT Generalisation Design Proposal](https://www.notion.so/chainlink/Design-Proposal-Chain-Agnostic-Head-Tracker-e7bc15a728ff44829c912b7c5aff8b62#d916686976024bbebdb996f246af6e9d)

### CODE REVIEW

File: common/headtracker/types/config.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package types
+
+import "time"
+
+type HtrkConfig interface {
```
Comment (jmank88) [inline common/headtracker/types/config.go:5]: Is this just a `Config`? given the `headtracker/types` context?
```suggestion
type Config interface {
```

File: core/chains/evm/headtracker/config.go [hunk 2]
```diff
@@ -12,3 +16,31 @@ type Config interface {
 	EvmHeadTrackerMaxBufferSize() uint32
 	EvmHeadTrackerSamplingInterval() time.Duration
 }
+
+type EvmHtrkConfig types.HtrkConfig
```
Comment (jmank88) [inline core/chains/evm/headtracker/config.go:20]: Isn't this just a `Config` too?
```suggestion
type Config types.Config
```

File: common/headtracker/types/config.go (hunk 1 above)
Comment (ExtremelySunnyYK) [reply common/headtracker/types/config.go:5]: Thanks for pointing that out. That makes more sense

File: core/chains/evm/headtracker/config.go (hunk 2 above)
Comment (ExtremelySunnyYK) [reply core/chains/evm/headtracker/config.go:20]: thanks for the note. Will be refactoring that out as we already have headtracker context

Comment (ExtremelySunnyYK) [reply core/chains/evm/headtracker/config.go:20]: However i think that having `EVM` in the Config name could still be useful especially as that config is evm specific.

Comment (jmank88) [reply core/chains/evm/headtracker/config.go]: > However i think that having `EVM` in the Config name could still be useful especially as that config is evm specific.

But is it used from any context that _isn't_ EVM specific?

File: core/chains/evm/headtracker/head_listener_test.go [hunk 3]
```diff
@@ -105,7 +105,7 @@ func Test_HeadListener_NotReceivingHeads(t *testing.T) {
 	evmcfg := evmtest.NewChainScopedConfig(t, cfg)
 	evmcfg.BlockEmissionIdleWarningThreshold()
 	chStop := make(chan struct{})
-	hl := headtracker.NewHeadListener(lggr, ethClient, evmcfg, chStop)
+	hl := headtracker.NewHeadListener(lggr, ethClient, headtracker.NewEvmConfig(evmcfg), chStop)
```
Comment (jmank88) [inline core/chains/evm/headtracker/head_listener_test.go:108]: If you drop `EVM` from `Config`, then worst case is that callers have to alias the package:
```go
	hl := headtracker.NewHeadListener(lggr, ethClient, evmheadtracker.NewConfig(evmcfg), chStop)
```
But it seems like everything can just do `headtracker.NewConfig`, no?

Comment (ExtremelySunnyYK) [reply core/chains/evm/headtracker/head_listener_test.go:108]: Thanks for the notes - dropping the EVM from config makes sense. I think `NewWrappedConfig` sounds like it is clearer.

**Review by jmank88 — APPROVED** [review]

**Review by prashantkumar1982 — APPROVED** [review]

**Pull request merged**